### ISSUE TITLE
Support dynamically adding children to `layout_column_wrap()`

### ISSUE THREAD

**Issue by gadenbuie** [rstudio/bslib#561]
Follow up to #559. This might be an advanced usage, but because `layout_column_wrap()` wraps each child element in a set of fill-carrying containers, it's less straight-forward than I expected to dynamically add new elements to an existing `layout_column_wrap()`.

The use case would be to set up a wrapping container that will hold an arbitrary number of elements and add to them using `insertUI()` or something similar. `layout_column_wrap()` is a good candidate for this because you can effectively give some parameters to the wrapping and sizing of child elements even if you don't know exactly how many will be on the page in advance.

The issue is that children of `layout_column_wrap()` are wrapped in

```html
<div class="html-fill-container">
  <div class="html-fill-item html-fill-container">
    <!-- CHILD CONTENTS -->
  </div>
</div>
```

so a naive implementation that just adds new elements to `layout_column_wrap()` might not work as expected.

One option would be to provide a `layout_column_wrap_item()` function, or something more generic, e.g. `item_wrapped()`. I'm not sure. We do now have `as_fill()`, `as_fillable()` and `as_fill_carrier()` functions, but these aren't great either as they each add a layer of nesting.

```r
as_fillable(
  div(
    as_fill_carrier(
      div(
        HTML("<!-- CHILD CONTENTS -->")
      )
    )
  )
)
#> <div class="html-fill-container">
#>   <div class="html-fill-container html-fill-item">
#>     <!-- CHILD CONTENTS -->
#>   </div>
#> </div>
```

**Comment by cpsievert** [comment]
I can see how this is a problem for `insertUI()`, but I don't think it's a problem for `layout_column_wrap(..., uiOutput("id", fill = TRUE))`?

**Comment by gadenbuie** [comment]
> I can see how this is a problem for `insertUI()`, but I don't think it's a problem for `layout_column_wrap(..., uiOutput("id", fill = TRUE))`?

Yeah, I was specifically thinking about adding children to the `layout_column_wrap()`. If you wanted to dynamically adjust the number of elements, currently moving the `layout_column_wrap()` into the `renderUI()` output is probably best.

This has me wondering: are the nested `.html-fill-container .html-fill-container.html-fill-item` divs required?

**Comment by thekangaroofactory** [comment]
Ah, that's exactly what I was looking for - thank you :)
I'm working on a project where I would like to avoid computing again the whole list of (many) cards when I "just" need to add one.

I will wrap my content as mentioned above and use `insertUI()`, but yeah it would be nice to have a dedicated function for that.
